src/equations/bbm_bbm_1d.jl

```diff
@@ -241,10 +256,18 @@ function rhs!(du_ode, u_ode, t, mesh, equations::BBMBBMEquations1D, initial_cond
 
     @timeit timer() "source terms" calc_sources!(dq, q, t, source_terms, equations, solver)
 
-    @timeit timer() "deta elliptic" deta[:]=invImD2n * deta
+    # To use the in-place version `ldiv!` instead of `\`, we need temporary arrays
+    # since `deta` and `dv` are not stored contiguously
+    @timeit timer() "deta elliptic" begin
+        tmp1[:] = deta
+        ldiv!(tmp3, invImD2n, tmp1)
+        deta[:] = tmp3
+    end
     @timeit timer() "dv elliptic" begin
-        dv[2:(end - 1)] = invImD2d * dv[2:(end - 1)]
+        tmp2[:] = dv
+        ldiv!(tmp4, invImD2d, tmp2[2:(end - 1)])
         dv[1] = dv[end] = zero(eltype(dv))
+        dv[2:(end - 1)] = tmp4
```

Comment:
And here

Reply:
Here I need another temporary vector because it has a different size (only N - 2 instead of N).